### PR TITLE
Return only Tags + DefaultValue + Declarative pipeline

### PR DESCRIPTION
**Pull request by Reuuke**
Adds support to return only Tags
Adds support to define defaultValue
Support image-tag-parameter-plugin in the declarative pipeline

### CODE REVIEW

**Comment by maduma** [comment]
return only tag is a good idea

Maybe the pugins should set a second variable ending with _TAG bye default
When using imageTag name: 'DOCKER_IMAGE'
DOCKER_IMAGE and DOCKER_IMAGE_TAG

**Comment by h1dden-da3m0n** [comment]
> return only tag is a good idea
> 
> Maybe the pugins should set a second variable ending with _TAG bye default
> When using imageTag name: 'DOCKER_IMAGE'
> DOCKER_IMAGE and DOCKER_IMAGE_TAG

Mind me asking if this even possible?
I mean I have not seen a parameter plugin that exports 2 parameters from one?

Otherwise love the idea! (I mostly need to split the result of the current version into image name and tag for my use-case)

**Comment by Reuuke** [comment]
> return only tag is a good idea
> 
> Maybe the pugins should set a second variable ending with _TAG bye default
> When using imageTag name: 'DOCKER_IMAGE'
> DOCKER_IMAGE and DOCKER_IMAGE_TAG

This should be a hidden or visible variable?
DOCKER_IMAGE = image + ":"+ tag? or only image name? Should include repository name? 
DOCKER_IMAGE_TAG = only tag

**Comment by maduma** [comment]
I would say a hidden variable.
DOCKER_IMAGE = image + ":"+ tag
DOCKER_IMAGE_TAG = only tag

**Comment by h1dden-da3m0n** [comment]
Okay, I had some time and played around a bit with this idea of additional environment exposure.
To my surprise, it was quite a lot simpler than I imagined this. See my [tagSplit](https://github.com/h1dden-da3m0n/image-tag-parameter-plugin/tree/tagSplit) branch for a POC.
I could open a PR with this or if @Reuuke wants to take this and merge it into their PR would be fine with me too.
Just let me know what to do :wink: 

---

What did I do in that branch:
* added a customer ParameterValue type that exposes 2 additional environment variables (in total 3)
    * `parameterName` the image+tag (e.g.: *openjdk:11-slim*)
    * `parameterName_IMAGE` the image name only (e.g.: *openjdk*)
    * `parameterName_TAG` the tag of the image only (e.g.: *11-slim*)
* add a copy of the StringParameterValue value.jelly so that this new ParameterValue can be rendered in the parameter view of a build job
* some minor tweaks (add .editorconfig and move sort location)

**Comment by h1dden-da3m0n** [comment]
Oh and yes I am aware and sure that this could have been handled much more elegantly than my POC.
So if you have suggestions I am all ear or if you want to use it maybe give it another sanity/style check.

File: src/main/java/io/jenkins/plugins/luxair/ImageTag.java [hunk 1]
```diff
@@ -20,14 +20,19 @@
...
-            .map(tag -> image + ":" + tag)
-            .collect(Collectors.toList());
+        if (returnOnlyTags){
+            return tags.stream().filter(tag -> tag.matches(filter))
+                .collect(Collectors.toList());
+        } else {
+            return tags.stream().filter(tag -> tag.matches(filter))
+                .map(tag -> image + ":" + tag)
+                .collect(Collectors.toList());
+        }
```
Comment (h1dden-da3m0n) [inline src/main/java/io/jenkins/plugins/luxair/ImageTag.java:35]:
```java
return tags.stream().filter(tag -> tag.matches(filter))
            .map(tag -> returnOnlyTags ? image + ":" + tag : tag)
            .collect(Collectors.toList());
```
I think that would be a bit easier to read :wink:

**Comment by h1dden-da3m0n** [comment]
Okay, as you can see by the mention of this PR in my latest commit, I have incorporated some of the code of this PR into my POC tagSplit branch.
I did this to get the rebuild-plugin to behave and select the previously selected value.
If @Reuuke is okay with it and @maduma is okay with my "solution" to export additional envVars to export the image tag on itself, then I would open an alternative PR to this PR with what is currently in my [tagSplit](https://github.com/h1dden-da3m0n/image-tag-parameter-plugin/tree/tagSplit) branch.

**Comment by Reuuke** [comment]
> tagSplit

Yes, I am completely satisfied with your solution.

**Comment by h1dden-da3m0n** [comment]
> > tagSplit
> 
> Yes, I am completely satisfied with your solution.

I was more asking for your permission to use the parts of your could, but I am happy you like my solution :grin:

**Pull request closed without merging**